src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -80,20 +84,36 @@ private void processBlob(String blobName, String containerName) {
                 leaseClient.acquireLease(60);
                 byte[] rawBlob = tryToDownloadBlob(blobClient);
 
-                // target storage account will be retrieved from configuration when Crime blob processing is supported
-                dispatcher.dispatch(blobName, rawBlob, containerName, TargetStorageAccount.BULKSCAN);
-                UUID envelopeId = envelopeRepository.insert(createNewEnvelope(
-                    blobName,
-                    containerName,
-                    blobCreationDate
-                ));
-
-                logger.info(
-                    "Finished processing {} from {} container. New envelope ID: {}",
-                    blobName,
-                    containerName,
-                    envelopeId
-                );
+                // Verify Zip signature
+                boolean validSignature = signatureVerifier.verifyZipSignature(blobName, containerName, rawBlob);
+
+                if (validSignature) {
+                    // target storage account will be retrieved from configuration
+                    // when Crime blob processing is supported
+                    dispatcher.dispatch(blobName, rawBlob, containerName, TargetStorageAccount.BULKSCAN);
+                    UUID envelopeId = envelopeRepository.insert(createNewEnvelope(
+                        blobName,
+                        containerName,
+                        blobCreationDate,
+                        Status.DISPATCHED
+                    ));
+
+                    logger.info(
+                        "Finished processing {} from {} container. New envelope ID: {}",
+                        blobName,
+                        containerName,
+                        envelopeId
+                    );
+                } else {
+                    envelopeRepository.insert(createNewEnvelope(
+                        blobName,
+                        containerName,
+                        blobCreationDate,
+                        Status.REJECTED
+                    ));
+                    // TODO move to rejected container
```

Comment:
this comment can be removed, moving to rejected container will not happen here, there will be a separate job moving such files.